audinterface/core/utils.py

```diff
@@ -137,21 +137,23 @@ def read_audio(
     if root is not None and not os.path.isabs(file):
         file = os.path.join(root, file)
 
+    sampling_rate = af.sampling_rate(file)
+
     if start is None or pd.isna(start):
         offset = 0
     else:
-        offset = start.total_seconds()
+        offset = to_samples(start, sampling_rate)
 
     if end is None or pd.isna(end):
         duration = None
     else:
-        duration = end.total_seconds() - offset
+        duration = to_samples(end, sampling_rate) - offset
 
     signal, sampling_rate = af.read(
         audeer.safe_path(file),
         always_2d=True,
-        offset=offset,
-        duration=duration,
+        offset=str(offset),  # str() to enforce samples as unit
+        duration=str(duration),
     )
```

Comment:
The offset/duration was previously given in seconds and now is given as number of samples.
I saw that audiofile.read and respectively audmath.duration_in_seconds treats integers as seconds and strings as samples.
I wonder what the deeper sense behind it is? Without deeper knowledge it looks a bit hacky to me.

Reply:
The idea behind the strange string option is to provide the possibility to use units like `'3 ms'` or provide samples without the need of having an additional `unit` argument.

The only problem was that we designed `audmath.duration_in_seconds()` in a way that is not compatible with how we can do it in `audiofile`:

> if `duration` is a float, integer or string without unit it is treated as seconds or if `sampling_rate` is provided as samples

The problem is in `audiofile.read()` we always have the sampling rate and hence we treat `'1000'` as representing 1000 samples instead of 1000 seconds. So yes, it's not perfect, but you also can just ignore it if you don't want to use it ;)

---

As we have now fixed `audiofile` we no longer need to change the code here in `audinterface`.

Reply:
I see, yes, all fine, i was just thinking in the first place, that this results in errors as we previously used seconds and now use samples and was quite surprised to see, that `1000` is handled differently than `"1000"`.